db/migrate/20231023144844_add_author_ref_to_posts.rb

```diff
@@ -0,0 +1,7 @@
+class AddAuthorRefToPosts < ActiveRecord::Migration[7.1]
+  def change
+    add_column :posts, :author_id, :integer
+    add_foreign_key :posts, :users, column: :author_id
+    add_index :posts, :author_id
+  end
```

Comment:
- [ ] [Optional] It would be better professionally if you add the foreign key as `author_id` between users and posts table like that instead:

```ruby
add_reference :posts, :author, foreign_key: { to_table: :users }, null: false
```